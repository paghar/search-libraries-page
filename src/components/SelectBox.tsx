import DownArrow from "./misc/svg/DownArrow";
import {ISelectBox, ISelectBoxItem} from "../utils/interface/misc";


const SelectBox = ((props:ISelectBox)=>{  

  const items = props?.items?.map((item:ISelectBoxItem)=>{        
    return <option key={item.value} value={item.value}>{item.text}</option>; 
  });

  return(       
    <div>
      <div className="relative inline-flex self-center">
        <DownArrow className="text-white bg-pink-700 absolute top-0 right-0 m-2 pointer-events-none p-2 rounded" />        
        <select 
          id={props?.id}
          value={props.selectedValue} 
          onChange={props.onChange}
          className="text-l font-bold rounded border-2 border-pink-700 text-gray-600 h-10 w-60 pl-5 pr-10 bg-white hover:border-gray-400 focus:outline-none appearance-none">
          {items}      
        </select>
      </div>
    </div>
     
  );
});
      
export default SelectBox;
      