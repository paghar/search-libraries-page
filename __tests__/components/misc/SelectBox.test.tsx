import React from "react";
import {render, fireEvent} from "@testing-library/react";
import SelectBox from "../../../src/components/misc/SelectBox";

const items = [
  {value: 1, text: "Option 1"},
  {value: 2, text: "Option 2"},
  {value: 3, text: "Option 3"},
];
const selectedValue = "2";
const onChange = jest.fn();

describe("Button Component", () => {
  const {container} = render(
    <SelectBox id="test-select" items={items} selectedValue={selectedValue} onChange={onChange} />
  );
  
  it("Renders Snapshot Buttons", () => {    
    expect(container).toMatchSnapshot();
  });
  
});

describe("SelectBox Component", () => {
  test("renders select box with provided items and handles selection change", () => {
    const {getByDisplayValue, getByText} = render(
      <SelectBox id="test-select" items={items} selectedValue={selectedValue} onChange={onChange} />
    );

    const selectBox = getByDisplayValue("Option 2"); // Ensure select box renders with the correct selected value
    expect(selectBox).toBeInTheDocument();

    // Ensure options are rendered
    items.forEach(item => {
      const option = getByText(item.text);
      expect(option).toBeInTheDocument();
    });

    // Simulate a selection change
    fireEvent.change(selectBox, {target: {value: "3"}});
    expect(onChange).toHaveBeenCalled();
  });
});
