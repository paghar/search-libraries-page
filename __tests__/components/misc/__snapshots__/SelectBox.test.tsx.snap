// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Snapshot Buttons 1`] = `
<div>
  <div>
    <div
      class="relative inline-flex self-center"
    >
      <svg
        class="text-white bg-pink-700 absolute top-0 right-0 m-2 pointer-events-none p-2 rounded"
        height="25px"
        version="1.1"
        viewBox="0 0 38 22"
        width="25px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          id="ZahnhelferDE—Design"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#ffffff"
            fill-rule="nonzero"
            id="ZahnhelferDE–Icon&Asset-Download"
            transform="translate(-539.000000, -199.000000)"
          >
            <g
              id="Icon-/-ArrowRight-Copy-2"
              transform="translate(538.000000, 183.521208)"
            >
              <polygon
                id="Path-Copy"
                points="33 5.38477631 33 31.3847763 29 31.3847763 28.999 9.38379168 7 9.38477631 7 5.38477631"
                transform="translate(20.000000, 18.384776) rotate(135.000000) translate(-20.000000, -18.384776) "
              />
            </g>
          </g>
        </g>
      </svg>
      <select
        class="text-l font-bold rounded border-2 border-pink-700 text-gray-600 h-10 w-60 pl-5 pr-10 bg-white hover:border-gray-400 focus:outline-none appearance-none"
        id="test-select"
      >
        <option
          value="1"
        >
          Option 1
        </option>
        <option
          value="2"
        >
          Option 2
        </option>
        <option
          value="3"
        >
          Option 3
        </option>
      </select>
    </div>
  </div>
</div>
`;
